Formula/petsc.rb

```diff
@@ -0,0 +1,43 @@
+class Petsc < Formula
+  desc "Scalable solution of models that use partial differential equations"
+  homepage "https://www.mcs.anl.gov/petsc/index.html"
+  url "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.9.0.tar.gz"
+  sha256 "a233e0d7f69c98504a1c3548162c6024f7797dde5556b83b0f98ce7326251ca1"
+
+  depends_on "gcc"
+  depends_on "hdf5"
+  depends_on "hwloc"
+  depends_on "metis"
+  depends_on "netcdf"
+  depends_on "open-mpi"
+  depends_on "scalapack"
+  depends_on "suite-sparse"
+
+  def install
+    system "./configure", "CC=mpicc",
+                          "CXX=mpicxx",
+                          "F77=mpif77",
+                          "FC=mpif90",
+                          "--prefix=#{prefix}",
+                          "--with-debugging=0",
```

Comment:
why

Reply:
The debug builds have expensive input validation and memory debugging diagnostics that are too expensive for production users. Is there a recommended way for homebrew to distributed separate debug and optimized libraries? (They are ABI compatible.)

Reply:
>Is there a recommended way for homebrew to distributed separate debug and optimized libraries? 

no.

Is a debug build actually default enabled?

Reply:
Yes, the debug variant is default because most users that are installing PETSc from source are developing code that sometimes has memory errors. We recommend using the debug variant for development and switching to optimized only after their code is working. I would estimate that about a third of mailing list support traffic is from bugs in user code that are detected by the debug variant.

But in the context of a package manager, it's common that PETSc is just a dependency for other libraries or applications, so optimized is the right default.

Reply:
ok